wiki/markdown/generate-cli.md

```diff
@@ -0,0 +1,190 @@
+# REMReM Generate CLI
+
+## Usage
+
+For using Eiffel REMReM Generate CLI self executing **generate-cli.jar** file should be executed in command line with parameters described below.
+
+```
+java -jar generate-cli.jar -h
+
+You passed help flag.
+usage: java -jar
+
+ -d,--debug                  enable debug traces
+
+ -f,--content_file           message content file
+
+ -h,--help                   show help
+
+ -json,--json_content        JSON content
+
+ -mp,--messaging_protocol    name of messaging protocol to be used, e.g. eiffelsemantics
+
+ -r,--response_file          file to store the response in, optional
+
+ -t,--message_type           message type, mandatory if -f or -json is given
+
+ -v,--list_versions          lists the versions of generate and all loaded protocols
+ ```
+
+
+**Generate templates for Eiffel REMReM Semantics to be used are available [here.](https://github.com/eiffel-community/eiffel-remrem-semantics)**
+
+The message must exist in a JSON file, for example **_eiffelactivityfinished.json_** might contain the following message:
+
+```
+{
+  "msgParams": {
+    "meta": {
+    "type": "EiffelActivityFinishedEvent",
+      "tags": [
+        "tag1",
+        "tag2"
+      ],
+      "source": {
+        "domainId": "example.domain",
+        "host": "host",
+        "name": "name",
+        "uri": "http://java.sun.com/j2se/1.3/",
+        "gav": {
```

Comment:
This event example still contains GAV and should be updated to Agen version.

Reply:
Hi Emelie,

Updated the suggested changes